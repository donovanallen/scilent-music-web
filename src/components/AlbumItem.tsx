// import { Album } from '@spotify/web-api-ts-sdk';
import Image from 'next/image';
import Link from 'next/link';
import React from 'react';
import { BiAlbum } from 'react-icons/bi';

import Pill from '@/components/Pill';

interface AlbumItemProps {
  id?: string;
  name?: string;
  image?: string | any;
  timestamp?: string | Date;
  type?: string;
  // album?: Album | any;
  // artist?: string | any;
  // className?: string;
}

const AlbumItem: React.FC<AlbumItemProps> = ({
  id,
  name,
  image,
  type,
  timestamp,
  // album,
  // artist,
  // className,
}) => {
  const imagePath = image;

  return (
    <Link href={`/release/${id}`}>
      <div className='rounded-md overflow-hidden bg-neutral-900 cursor-pointer hover:opacity-75 hover:text-brand-light transition'>
        <div className='relative aspect-square rounded-md w-full bg-neutral-700 animated-underline'>
          {imagePath ? (
            <Image
              src={imagePath}
              alt={name + ' image'}
              fill
              className='aspect-square object-cover hover:pb-1 transition'
            />
          ) : (
            <BiAlbum size={64} className='m-auto h-full text-dark' />
          )}
          {type && (
            <Pill
              text={type}
              className='absolute bottom-2 right-2 bg-neutral-800/90'
            />
          )}
        </div>
        <div className='relative p-3 w-full overflow-clip'>
          <h4 className='subtitle font-normal truncate'>{name}</h4>
          <p className='subtitle font-normal text-neutral-500 truncate'>
            {new Date(timestamp as Date).getFullYear()}
          </p>
        </div>
      </div>
    </Link>
  );
};

export default AlbumItem;
