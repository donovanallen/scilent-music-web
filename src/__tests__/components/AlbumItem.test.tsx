import { render } from '@testing-library/react';
import React from 'react';

import AlbumItem from '@/components/AlbumItem';

describe('AlbumItem', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<AlbumItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});
