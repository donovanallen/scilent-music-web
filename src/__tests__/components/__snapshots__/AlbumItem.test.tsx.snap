// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumItem renders correctly 1`] = `
<DocumentFragment>
  <a
    aria-label="Go to album undefined"
    href="/release/undefined"
  >
    <div
      class="rounded-md overflow-hidden bg-neutral-900 hover:text-brand-light hover:border-brand-light hover:border transition"
    >
      <div
        class="relative aspect-square rounded-md w-full bg-neutral-700 animated-underline"
      >
        <svg
          class="m-auto h-full text-dark"
          fill="currentColor"
          height="64"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="64"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11.99"
            cy="11.99"
            r="2.01"
          />
          <path
            d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm0 18a8 8 0 1 1 8-8 8 8 0 0 1-8 8z"
          />
          <path
            d="M12 6a6 6 0 0 0-6 6h2a4 4 0 0 1 4-4z"
          />
        </svg>
      </div>
      <div
        class="relative p-3 w-full overflow-clip"
      >
        <p
          class="subtitle font-normal truncate"
        />
        <p
          class="subtitle font-normal text-neutral-500 truncate"
        />
      </div>
    </div>
  </a>
</DocumentFragment>
`;
